src/main/kotlin/io/zerobase/smarttracing/features/notifications/NotificationsFeatureFactory.kt

```diff
@@ -0,0 +1,14 @@
+package io.zerobase.smarttracing.features.notifications
+
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.google.inject.Injector
+import io.dropwizard.setup.Environment
+import io.zerobase.smarttracing.config.AppConfig
+import io.zerobase.smarttracing.features.FeatureFactory
+
+@JsonTypeName("notifications")
+class NotificationsFeatureFactory(override val enabled: Boolean) : FeatureFactory {
+    override fun build(env: Environment, config: AppConfig, injector: Injector) {
+        // nothing to do, guice is handling it all
```

Comment:
does it have to override the function then if it's not doing anything?

Reply:
yes, it's the nature of an interface implementation. the factory has to exist so it can be in the list for the configuration, but it's not actually registering rest apis or anything else in this phase of the lifecycle.